greet_app/greet.py

```diff
@@ -1,25 +1,42 @@
 def greet_with_many_names(names):
-    return f"Hello, {', '.join(names[:-1])} and {names[-1]}."
+    return f"Hello, {', '.join(names[:-1])}{' and ' if len(names) > 1 else ''}{names[-1]}."
```

Comment:
now this method and method below differ very slightly. could you resolve duplication?